src/primitive_core.cc

```diff
@@ -69,15 +69,23 @@ namespace toit {
 MODULE_IMPLEMENTATION(core, MODULE_CORE)
 
 PRIMITIVE(write_string_on_stdout) {
-  ARGS(cstring, message, bool, add_newline);
-  fprintf(stdout, "%s%s", message, add_newline ? "\n" : "");
+  ARGS(Blob, message, bool, add_newline);
+  auto bytes = message.address();
+  for (int i = 0; i < message.length(); i++) {
+    putchar(bytes[i]);
+  }
+  if (add_newline) putchar('\n');
   fflush(stdout);
   return process->null_object();
 }
 
 PRIMITIVE(write_string_on_stderr) {
-  ARGS(cstring, message, bool, add_newline);
-  fprintf(stderr, "%s%s", message, add_newline ? "\n" : "");
+  ARGS(Blob, message, bool, add_newline);
+  auto bytes = message.address();
+  for (int i = 0; i < message.length(); i++) {
+    putchar(bytes[i]);
```

Comment:
How is this stderr? This looks like a pretty significant regression.

I'd take something like my suggestion above and maybe put it in a static helper function parameterized with the stream.

Also, consider making both member variables of `Blob` const like:

```
 private:
  const uint8* const address_;
  const word length_;
```

Reply:
Good catch...
I will update the blob fields in a separate PR.